lib/perlfaq3.pod

```diff
@@ -340,214 +340,78 @@ L<http://www.cpan.org/authors/id/T/TO/TOMC/scripts/toms.exrc.gz>
 
-L<http://www.gnu.org/software/emacs/windows/ntemacs.html>
+=item L<BBEdit|http://www.barebones.com/> or TextWrangler (macOS)
 
-=item MicroEMACS
+=item L<ConTEXT|http://www.contexteditor.org/> (Windows)
 
-L<http://www.microemacs.de/>
+No LSP support.
 
-=item XEmacs
+=item L<Eclipse|http://e-p-i-c.sf.net/>
 
-L<http://www.xemacs.org/Download/index.html>
+=item L<Emacs|https://www.gnu.org/software/emacs/download.html>
 
-=item Jed
+Use cperl-mode.
 
-L<http://space.mit.edu/~davis/jed/>
+=item L<Geany|https://www.geany.org>
 
-=back
-
-or a vi clone such as
+=item L<IntelliJ IDEA|https://www.jetbrains.com/idea/>
 
-=over 4
+USe the L<Perl plugin from the Jet Brains marketplace|https://plugins.jetbrains.com/plugin/7796>.
 
-=item Vim
+Camelcade plugin provides Perl5 support in IntelliJ IDEA and other JetBrains IDEs:
+L<Perl5-IDEA|https://github.com/Camelcade/Perl5-IDEA/>.
 
-L<http://www.vim.org/>
+IDEA Ultimate includes LSP support.
 
-=item Vile
+=item L<Kate|https://kate-editor.org>
 
-L<http://invisible-island.net/vile/vile.html>
+=item L<Notepad++|https://notepad-plus-plus.org> (Windows)
 
-=back
-
-The following are Win32 multilanguage editor/IDEs that support Perl:
-
-=over 4
+No LSP support.
 
-=item MultiEdit
+=item L<Sublime Text|https://www.sublimetext.com/>
 
-L<http://www.MultiEdit.com/>
+=item L<Textpad|http://www.textpad.com/> (Windows)
 
-=item SlickEdit
+No LSP support.
 
-L<http://www.slickedit.com/>
+=item L<UltraEdit|http://www.ultraedit.com/>
 
-=item ConTEXT
+=item vi and L<vim|http://www.vim.org/>, L<NeoVim|https://neovim.io>
 
-L<http://www.contexteditor.org/>
+There are several other forks or customizations of the vi line of editors.
+You might also like L<Vim::Debug::Manual> and
+L<coc-perl|https://github.com/perl-ide/coc-perl>.
 
-=back
-
-There is also a toyedit Text widget based editor written in Perl
-that is distributed with the Tk module on CPAN. The ptkdb
-( L<http://ptkdb.sourceforge.net/> ) is a Perl/Tk-based debugger that
-acts as a development environment of sorts. Perl Composer
-( L<http://perlcomposer.sourceforge.net/> ) is an IDE for Perl/Tk
-GUI creation.
-
-In addition to an editor/IDE you might be interested in a more
-powerful shell environment for Win32. Your options include
-
-=over 4
-
-=item bash
-
-from the Cygwin package ( L<http://cygwin.com/> )
-
-=item zsh
-
-L<http://www.zsh.org/>
-
-=back
-
-Cygwin is covered by the GNU General Public
-License (but that shouldn't matter for Perl use). Cygwin
-contains (in addition to the shell) a comprehensive set
-of standard Unix toolkit utilities.
-
-=over 4
+=item L<Visual Studio Code|https://code.visualstudio.com/>
 
-=item BBEdit and TextWrangler
+Install the L<Perl plugin|https://marketplace.visualstudio.com/items?itemName=richterger.perl>
+from the Visual Studio Marketplace.
 
-are text editors for OS X that have a Perl sensitivity mode
-( L<http://www.barebones.com/> ).
+=item L<WebStorm|https://www.jetbrains.com/webstorm/>
```

Comment:
Webstorm as well as other jetbrains IDEs  (RubyMine, PhpStorm, PyCharm, etc. see [supported products](https://plugins.jetbrains.com/plugin/7796-perl/versions/stable/412532)) supporting perl only with perl plugin mentioned above. Well, you may have a textmate highlighting without it, but that's it. Any code intelligence, profiling, coverage, debugging and so on provided by the plugin.